File: Activity_and_Hooks/app/src/main/java/com/example/activity_and_hooks/MainActivity.java
package com.example.activity_and_hooks;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {


    public void onSecondPage(View view){
//        Intent secondIntent = new Intent(this, SecondActivity.class);
//        startActivity(secondIntent);
//        Explicit Intent

//        Implicit
        Intent implicitIntent = new Intent();
        implicitIntent.setAction(Intent.ACTION_SEND);
        implicitIntent.putExtra("text", "Custom text message");
        implicitIntent.setType("text/plain"); // MIME Type


        Intent windowChooser = Intent.createChooser(implicitIntent, "Application List");

        if( implicitIntent.resolveActivity(getPackageManager()) != null ){
            startActivity(windowChooser);
        }
        else {
            Toast.makeText(this, "Fail", Toast.LENGTH_SHORT).show();
        }

    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.i("LIFE_CYCLE", "MainActivity.onCreate");

//        Button button = (Button) findViewById(R.id.secondPageButton);
//
//        button.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                onSecondPage(v);
//            }
//        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i("LIFE_CYCLE", "MainActivity.onStart");

    }
    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i("LIFE_CYCLE", "MainActivity.onDestroy");
    }
    @Override
    protected void onPause() {
        super.onPause();
        Log.i("LIFE_CYCLE", "MainActivity.onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i("LIFE_CYCLE", "MainActivity.onStop");
    }
    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i("LIFE_CYCLE", "MainActivity.onRestart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i("LIFE_CYCLE", "MainActivity.onResume");
    }
}

File: Email_Password_Firebase_G29/app/src/main/java/com/example/email_password_firebase_g29/MainActivity.kt
package com.example.email_password_firebase_g29

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {
    private lateinit var signOutButton: Button
    private lateinit var updatePasswordButton: Button
    private lateinit var displayName: TextView

    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        auth = FirebaseAuth.getInstance()
        signOutButton = findViewById(R.id.signOutButton)
        displayName = findViewById(R.id.displayName)
        updatePasswordButton = findViewById(R.id.updatePasswordButton)

        signOutButton.setOnClickListener {
            signOut()
        }

        updatePasswordButton.setOnClickListener {
            val intent = Intent(this, UpdatePasswordActivity::class.java)
            startActivity(intent)
        }

        displayName.text = auth.currentUser?.displayName
    }

    private fun signOut(){
        auth.signOut()
        val intent = Intent(this, RegisterActivity::class.java)
        startActivity(intent)
        finish()
    }
}